cloud/wfm/src/JsonSchema.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2016 Richard Adams (https://github.com/enriched)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+// retrieved from https://gist.github.com/enriched/c84a2a99f886654149908091a3183e15
+
+interface JsonSchema {
```

Comment:
The `JsonSchema.ts` file for compile-time checking and ajv for runtime validation.

I was thinking that by requiring a schema for configuring steps we could provide more value out of the box:

- generating forms for configuring a Task on the portal app
- validating the data coming from the configuration form
- in general provide a pattern for how tasks should be configured from the ui

We can defer this for later I guess, always a tradeoff between the base interfaces having more features and being 'customizable'.